File: solutions/1089-Duplicate-Zeros.js
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
const duplicateZeros = function(arr) {
  let originalSize = arr.length;
  let i = 0;

  while(arr[i] != undefined) {
    if (arr[i] == 0) {
      arr.splice(i, 0, 0);
      i += 2;
    } else {
       i++;
    }
  }
  
  arr.splice(originalSize, arr.length - originalSize);
};

File: solutions/1089-Duplicate-Zeros.cpp
class Solution {
public:
  void duplicateZeros(vector<int>& arr) {
    auto it = arr.begin();
    int count {0};
    while (true) {
      if(it + 1 == arr.end()) {break;}
      if (*it == 0) {
        count++;
        it = arr.insert(it + 1, 0);
      }
    
      it++;
    }
    
    for (int i = 0; i < count; ++ i)
      arr.pop_back();
  }
};
